src/main/java/org/apache/bcel/generic/LDC.java

```diff
@@ -109,6 +111,9 @@ public Object getValue(final ConstantPoolGen cpg) {
             final int nameIndex = ((org.apache.bcel.classfile.ConstantClass) c).getNameIndex();
             c = cpg.getConstantPool().getConstant(nameIndex);
             return Type.getType(Type.internalTypeNameToSignature(((org.apache.bcel.classfile.ConstantUtf8) c).getBytes()));
+        case org.apache.bcel.Const.CONSTANT_Dynamic:
+            // Really not sure what to return here, maybe a BootstrapMethod instance but how do we get it?
+            return c;
```

Comment:
Any guidance or pointer would be very helpful here: my understanding is that the _value_ should be the return of the invocation of a bootstrap method, so it could be anything.
I'm not sure what's the best way for BCEL to convey this information because even the type of this value does not seem to be available here

Reply:
The value of a constant dynamic is only known at runtime, so I guess it can only return an arbitrary value ?

Reply:
Yes, I was wondering if there's already something in BCEL to account for values known at runtime.
For other constant types `LDC.getValue()` returns the actual value (or a `Type` in case it's a class), so returning the constant itself might be a bit misleading here

Reply:
I suppose, maybe later, we can invent a `ConstantDynamic` proxy class where you'd say here `new ConstantDynamic(c)` for a later handling, possibly by BCEL call sites themselves. I don't see how an `LDC`, `ConstantPoolGen` and `Attribute` containing the needed data intersect from here.